Dockerfile.coverage

```
@@ -0,0 +1,124 @@
+  llvm-${LLVM_V} \
+  llvm-${LLVM_V}-dev \
+  clang-${LLVM_V} \
+  cpio \
+  git \
+  build-essential \
+  libtool \
+  autotools-dev \
+  automake \
+  cmake \
+  pkg-config \
+  bsdmainutils \
+  openssh-client \
+  libcapstone-dev \
+  python3 \
+  libzstd-dev \
+  tmux \
+  vim \
+  gnuplot
+
+RUN apt-get install -y libsqlite3-dev
+RUN apt-get install -y libevent-dev libboost-dev
+
+# Install rust and tools
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+RUN rustup install nightly && rustup default nightly
+
+# Clone AFLplusplus and build
+ENV LLVM_CONFIG=llvm-config-${LLVM_V}
+RUN git clone https://github.com/AFLplusplus/AFLplusplus
+RUN cd AFLplusplus && make PERFORMANCE=1 install -j$(nproc --ignore 1)
+
+# ------ Build Bitcoin Core ------
+
+# Build Bitcoin Core
+ARG OWNER=bitcoin
+ARG REPO=bitcoin
+ARG BRANCH=master
+RUN git clone --depth 1 --branch $BRANCH https://github.com/$OWNER/$REPO.git
+
+ENV CC=$PWD/AFLplusplus/afl-clang-fast
+ENV CXX=$PWD/AFLplusplus/afl-clang-fast++
+ENV LD=$PWD/AFLplusplus/afl-clang-fast
+
+ENV SOURCES_PATH=/tmp/bitcoin-depends
+RUN make -C bitcoin/depends NO_QT=1 NO_ZMQ=1 NO_USDT=1 download-linux SOURCES_PATH=$SOURCES_PATH
+# Keep extracted source
+RUN sed -i --regexp-extended '/.*rm -rf .*extract_dir.*/d' ./bitcoin/depends/funcs.mk && \
+    make -C ./bitcoin/depends DEBUG=1 NO_QT=1 NO_BDB=1 NO_ZMQ=1 NO_USDT=1 \
+      SOURCES_PATH=$SOURCES_PATH \
+      AR=llvm-ar-${LLVM_V} NM=llvm-nm-${LLVM_V} RANLIB=llvm-ranlib-${LLVM_V} STRIP=llvm-strip-${LLVM_V} \
+      -j$(nproc)
+
+COPY ./target-patches/bitcoin-core-rng.patch bitcoin/
+
+RUN cd bitcoin/ && \
+      git apply bitcoin-core-rng.patch
+
+RUN cd bitcoin/ && cmake -B build_fuzz_cov \
+      --toolchain ./depends/$(./depends/config.guess)/toolchain.cmake \
+      -DAPPEND_CFLAGS="-fprofile-instr-generate -fcoverage-mapping" \
+      -DAPPEND_CXXFLAGS="-fprofile-instr-generate -fcoverage-mapping" \
+      -DAPPEND_LDFLAGS="-fprofile-instr-generate -fcoverage-mapping -fuse-ld=lld-${LLVM_V}" \
+      -DAPPEND_CPPFLAGS="-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
+
+RUN cmake --build bitcoin/build_fuzz_cov -j$(nproc) --target bitcoind
+
+ENV CC=
+ENV CXX=
+ENV LD=
+
+WORKDIR /fuzzamoto/fuzzamoto-nyx-sys
+COPY ./fuzzamoto-nyx-sys/Cargo.toml .
+COPY ./fuzzamoto-nyx-sys/src/ src/
+COPY ./fuzzamoto-nyx-sys/build.rs .
+
+WORKDIR /fuzzamoto/fuzzamoto
+COPY ./fuzzamoto/Cargo.toml .
+COPY ./fuzzamoto/src/ src/
+
+WORKDIR /fuzzamoto/fuzzamoto-cli
+COPY ./fuzzamoto-cli/Cargo.toml .
+COPY ./fuzzamoto-cli/src/ src/
+
+WORKDIR /fuzzamoto/fuzzamoto-scenarios
+COPY ./fuzzamoto-scenarios/Cargo.toml .
+COPY ./fuzzamoto-scenarios/bin/ bin/
+COPY ./fuzzamoto-scenarios/grammars/ grammars/
+
+WORKDIR /fuzzamoto
+COPY ./Cargo.toml .
+RUN mkdir .cargo && cargo vendor > .cargo/config
+
+ENV BITCOIND_PATH=/bitcoin/build_fuzz_cov/bin/bitcoind
+RUN cargo build --workspace --verbose --features reduced_pow --release
```

```suggestion
RUN cargo build --package fuzzamoto-scenarios --package fuzzamoto-cli --bins --verbose --features "fuzzamoto/reduced_pow" --release
```

So we only build `fuzzamoto-cli` and the scenario bins.